homeassistant/components/comfoconnect/fan.py

```diff
@@ -30,28 +30,30 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the ComfoConnect fan platform."""
     ccb = hass.data[DOMAIN]
 
-    add_entities([ComfoConnectFan(hass, name=ccb.name, ccb=ccb)], True)
+    add_entities([ComfoConnectFan(ccb.name, ccb)], True)
 
 
 class ComfoConnectFan(FanEntity):
     """Representation of the ComfoConnect fan platform."""
 
-    def __init__(self, hass, name, ccb: ComfoConnectBridge) -> None:
+    def __init__(self, name, ccb: ComfoConnectBridge) -> None:
         """Initialize the ComfoConnect fan."""
 
         self._ccb = ccb
         self._name = name
 
-        # Ask the bridge to keep us updated
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
         self._ccb.comfoconnect.register_sensor(SENSOR_FAN_SPEED_MODE)
```

Comment:
This call doesn't seem safe to run in the event loop. It uses a sync socket and blocking queue, with a 5 second timeout.

We need to schedule this call on the executor thread pool with `self.hass.async_add_executor_job`.

Reply:
I changed this.